include/CL/sycl/platform/detail/platform_tail.hpp

```diff
@@ -39,7 +39,10 @@ platform::get_devices(info::device_type device_type) const {
 
   // Return the devices with the good criterion according to the selector
   std::copy_if(devices.begin(), devices.end(), std::back_inserter(sd),
-               [&](const device &e) { return s(e) >= 0; });
+               [&](const device &e) {
+                 return s(e) >= 0 &&
+                   e.get_platform().get_info<info::platform::name>() == this->get_info<info::platform::name>();
```

Comment:
Would not be better to compare the platforms themselves instead of their name?

The line is tool long otherwise... 

Anyway, I do not think this is the good way to go. See above.
Perhaps it is also possible to fuse the 2 loops?